testnet/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -585,10 +585,41 @@ impl UTXO {
     }
 
     pub fn get_sat_amount(&self) -> u64 {
-        let res = self.amount * 10e7;
-        res as u64
+        UTXO::serialized_btc_to_sat(&self.amount)
     }
 
+    pub fn serialized_btc_to_sat(amount: &str) -> u64 {
+        let comps: Vec<&str> = amount.split(".").collect();
+        match comps[..] {
+            [lhs, rhs] => {
+                let base: u64 = 10;
+                let sat_decimals = 7;
+                let btc_to_sat = base.pow(7);
+                let btc = lhs.parse::<u64>().expect("Invalid amount");
+                let mut amount = btc * btc_to_sat;
+                if rhs.len() > sat_decimals { 
+                    panic!("Unexpected amount of decimals");
+                }
+                let rhs = format!("{:0<width$}", rhs, width = sat_decimals);
```

Comment:
Instead of creating a string and doing a second parse over it, you might want to do something like:

```
let frac_part = rhs.parse::<u64>().expect("Invalid amount");
assert!(rhs.len() <= 7, "Unexpected amount of decimals");
let sats = frac_part * base.pow(8 - rhs.len());
```

Reply:
I was creating a string for right padding the decimals, in case we don't want to panic and want to support values like `0.0001`. The suggested approach looks like a commitment to one specific version of bitcoin-core, on a format that is not consensus critical.
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1381/commits/ade5e904e8bbc9f0fd797ef2c004091a618ae387

Reply:
I don't think it does?  It only limits the RPC call to sending 7 decimal places or fewer.  But the `frac_part * base.pow(8 - rhs.len())` should ensure that the `rhs` substrings of `0.1`, `0.10, and `0.1000000` all parse to `10_000_000`.

Reply:
Yes, apologies.
I already had this condition in place 
```
if rhs.len() > 8
```
and by reading your snippet
```
assert!(rhs.len() <= 7)
```
I thought that you wanted to add a missing condition.
Time to get some rest :)
Patched with latest commit.